engine.go

```diff
@@ -24,6 +24,11 @@ type (
 		New(ctx context.Context, host HostCallHandler, guest []byte, config *ModuleConfig) (Module, error)
 	}
 
+	MeteringInfo struct {
```

Comment:
Since this is not an abstract, and possibly unnatural way, to provide metering info in go, what if instead the caller casts the concrete type when they need this? In other words, define this on the only engine that supports it. Then if metering becomes common and in a more natural way (vs unsafe pointers), there's still the ability to abstract it. However, abstracting with only one impl seems not the right time. SG?